### PR TITLE
fix: Scan for WebServlet annotation

### PR DESCRIPTION
**Pull request by mat3e**
Following this example I couldn't make `@WebServlet` annotation work. I reproduced the issue with test and fixed it following:
* https://stackoverflow.com/questions/77007560/missing-jetty-servlet-12-0-0-dependency
* https://stackoverflow.com/questions/68058157/jetty-11-doesnt-detect-jakarta-servlets
* https://happycoding.io/tutorials/java-server/embedded-jetty

Also, seems `web.xml` is not used at all (it points to Servlets 3.1) ~and there were some minor issues, like missing `@Override` I fixed.~

### CODE REVIEW

File: embedded/ee10-servlet-server/src/test/java/examples/EmbedMeTest.java [hunk 1]
```diff
@@ -29,33 +29,45 @@
...
         LifeCycle.stop(server);
     }
 
     @Test
-    public void testGetWelcome() throws IOException, InterruptedException
+    void testGetWelcome() throws IOException, InterruptedException
     {
-        HttpClient client = HttpClient.newBuilder().version(HttpClient.Version.HTTP_1_1).build();
-        HttpRequest request = HttpRequest.newBuilder()
+        var client = HttpClient.newBuilder().version(HttpClient.Version.HTTP_1_1).build();
```
Comment (joakime) [inline embedded/ee10-servlet-server/src/test/java/examples/EmbedMeTest.java]: No, keep the variable, don't fall back to `var`.
The use of `var` makes for bad and hard to follow examples. (the whole point of this repo)

File: embedded/ee10-servlet-server/src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -1,8 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
-  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
-		 http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
-  version="3.1">
```
Comment (joakime) [inline embedded/ee10-servlet-server/src/main/webapp/WEB-INF/web.xml:6]: Don't delete this, upgrade it to ee6 instead.

File: embedded/ee10-servlet-server/src/main/java/examples/EmbedMe.java [hunk 3]
```diff
@@ -51,6 +49,7 @@ public static Server newServer(int port)
         context.setBaseResource(baseResource);
         context.setContextPath("/");
         context.setWelcomeFiles(new String[]{"index.html", "welcome.html"});
+        context.setAttribute("org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern", ".*/");
```
Comment (joakime) [inline embedded/ee10-servlet-server/src/main/java/examples/EmbedMe.java]: Set the pattern properly, don't include 100% of the container.
As on Maven, this even includes all of the maven deps, and the maven plugins too.

In fact, the default pattern in ee10 should work out of the box.

**Review by joakime — CHANGES_REQUESTED** [review]
The addition of `jetty-ee10-annotation` dependency, and the test case, should be the only things this PR needs.

File: embedded/ee10-servlet-server/src/main/java/examples/EmbedMe.java (hunk 3 above)
Comment (mat3e) [reply embedded/ee10-servlet-server/src/main/java/examples/EmbedMe.java]: What should be the pattern? When I removed this line (using default?), test was failing. I changed to `.*/classes/*` now.

Comment (joakime) [reply embedded/ee10-servlet-server/src/main/java/examples/EmbedMe.java]: There are two patterns that impact how the webapp scanning operates.

1. The Container level (`org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern`) which scans for things that are not part of the webapp.
     By default, it will not scan the container, unless this pattern is defined.
2. The WebApp level (`org.eclipse.jetty.server.webapp.WebInfIncludeJarPattern`) which scans for things that belong specifically to the webapp.
     By default, this will scan all artifacts on the webapp (like `WEB-INF/lib/` and `classes/`) unless the pattern is defined.

For the container, you might need to support servlet, and jsp, and taglib.

For servlet only ...

``` java
webappContext.setAttribute(
    "org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern",
    ".*/jakarta.servlet-api-[^/]*\\.jar$");
```

For servlet, with JSP, and even Taglib support ...

``` java
webappContext.setAttribute(
    "org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern",
    ".*/jakarta.servlet-api-[^/]*\\.jar$|.*/jakarta.servlet.jsp.jstl-.*\\.jar$|.*/[^/]*taglibs.*\\.jar$");
```

Add more jar patterns for other jars that you might need from the container (which shouldn't be much).
Point is, don't open everything on the container for scanning, that's just wrong way to use the pattern, wrong for performance as you scan everything, wrong for security as you never want to open everything to the webapp, etc ...

Comment (joakime) [reply embedded/ee10-servlet-server/src/main/java/examples/EmbedMe.java]: The overwhelming majority of embedded use cases never scan, as that's an anti-pattern for embedded use cases (which focus on absolute control, and startup performance).

They instead define the webapp configuration entirely in code for quick startup.

This is the reason why those patterns are the way they are by default.

**Comment by joakime** [comment]
There are build errors with this latest code push.

Run `mvn spotless:apply` once and repush please.

**Comment by mat3e** [comment]
I applied Spotless, but when it comes to `ContainerIncludeJarPattern`, `".*/classes/*"` was the narrowest I found to pass my test expecting `TestServlet` to be registered 👀
